README
NAME
    Dist::Zilla::App::Command::lsplugins - Show all dzil plugins on your
    system, with descriptions

VERSION
    version 0.003001

SYNOPSIS
        dzil lsplugins # see a list of all plugins on your system
        dzil lsplugins --version # with versions!
        dzil lsplugins --sort    # sort them!
        dzil lsplugins --abstract # show their ABSTRACTs!
        dzil lsplugins --with=-FilePruner # show only file pruners
        dzil lsplugins --roles=dzil  # show all the dzil related role data!

METHODS
  "opt_spec"
    Supported parameters:

    *   "--sort"

        Sorting.

    *   "--no-sort"

        No Sorting ( Default )

    *   "--versions"

        Versions

    *   "--no-versions"

        No Versions ( Default )

    *   "--abstract"

        Show abstracts

    *   "--no-abstract"

        Don't show abstracts ( Default )

    *   "--roles=all"

        Show all roles, un-abbreviated.

    *   "--roles=dzil-full"

        Show only "dzil" roles, un-abbreviated.

    *   "--roles=dzil"

        Show only "dzil" roles, abbreviated.

    *   "--with=$ROLENAME"

        Show only plugins that "does($rolename)"

        ( A - prefix will be expanded to "Dist::Zilla::Role::" for
        convenience )

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-App-Command-lsplugins

0.003001 2014-10-31T16:44:00Z b85e47b

0.003000 2014-10-31T16:43:28Z 70666e2
 [Bugfix]
 - Avoid errors with utf8 loading for abstract parsing.
 - Avoid warnings with empty abstracts

 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +6 ↑2 -1 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -2
 - test: +1 (recommends: ↑2)

 [Internals]
 - remove utf8
 - support 5.006
 - drop App::Command loading Moose and MooseX::NonMoose

 [Metadata]
 - author = kentnl@cpan.org

0.002001 2014-08-15T21:14:50Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - tests augmented.
 - whitespace adjusted.

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - test: -3

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - update test::reportprereqs

0.002000 2014-08-01T08:09:39Z
 [00 Major]
 - Some changes that may affect hand-operated downstreams.

 [01 Toolkit]
 - Now uses ExtUtils::MakeMaker
 - "Latest" dependencies now only recommendations.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +5 ↓1 (recommends: +3)

0.001000 2013-11-12T02:17:10Z
 - First version.



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "Show all C<dzil> plugins on your system, with descriptions",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-App-Command-lsplugins",
  "dist_version" => "0.001001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::App::Command::lsplugins",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Util" => 0,
    "Module::Metadata" => 0,
    "Module::Runtime" => 0,
    "Moose" => 0,
    "MooseX::NonMoose" => 0,
    "Path::ScanINC" => 0,
    "Path::Tiny" => 0,
    "Try::Tiny" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4206",
  "Scalar::Util" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    un dzil
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.023.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Show all dzil plugins on your system, with descriptions",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-App-Command-lsplugins",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::App::Command::lsplugins",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Util" => 0,
    "Module::Metadata" => 0,
    "Module::Runtime" => 0,
    "Moose" => 0,
    "Path::ScanINC" => 0,
    "Path::Tiny" => 0,
    "Try::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.89"
  },
  "VERSION" => "0.003001",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::App" => 0,
  "Dist::Zilla::Util" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Metadata" => 0,
  "Module::Runtime" => 0,
  "Moose" => 0,
  "Path::ScanINC" => 0,
  "Path::Tiny" => 0,
  "Test::More" => "0.89",
  "Try::Tiny" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



